containerd-shim/process_solaris.go

```diff
@@ -0,0 +1,62 @@
+// +build solaris
+
+package main
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+// setPDeathSig is a no-op on Solaris as Pdeathsig is not defined.
+func setPDeathSig() *syscall.SysProcAttr {
+	return nil
+}
+
+// openIO opens the pre-created fifo's for use with the container
+// in RDWR so that they remain open if the other side stops listening
+func (p *process) openIO() error {
+	p.stdio = &stdio{}
+	var (
+		uid = p.state.RootUID
+	)
+	i, err := p.initializeIO(uid)
+	if err != nil {
+		return err
+	}
+	p.shimIO = i
+	// non-tty
```

Comment:
Terminal is supported but it just can't be passed in like on Linux. It will be created by the framework based on the pipes it is passed.
I didn't error out there because I didn't want to give the impression that tty semantics weren't supported..